src/Console/ModelsCommand.php

```diff
@@ -866,4 +869,44 @@ private function getClassKeyword(ReflectionClass $reflection)
 
         return $keyword;
     }
+
+    protected function checkForCustomLaravelCasts(string $type): ?string
+    {
+        if (!class_exists($type)) {
+            return $type;
+        }
+
+        $reflection = new \ReflectionClass($type);
+
+        if (!$reflection->implementsInterface(CastsAttributes::class)) {
+            return $type;
+        }
+
+        $methodReflection = new \ReflectionMethod($type, 'get');
+
+        return $this->getReturnTypeFromReflection($methodReflection);
+    }
+
+    protected function getReturnTypeFromReflection(\ReflectionMethod $reflection): ?string
+    {
+        $returnType = $reflection->getReturnType();
+
+        if (!$returnType) {
+            return $this->getReturnTypeFromDocBlock($reflection);
+        }
+
+        $type = $returnType instanceof \ReflectionNamedType
+            ? $returnType->getName()
+            : (string)$returnType;
+
+        if (!$returnType->isBuiltin()) {
+            $type = '\\' . $type;
+        }
+
+        if ($returnType->allowsNull()) {
+            $type .= '|null';
+        }
+
+        return $type;
```

Comment:
I get the impression I've seen the code somewhere 😄

Reply:
yep, took it from your pr, but inly becouse it is not merged yet